### PR TITLE
gpu_ctx_gl: reset depth/stencil write masks before calling glClear()

### PR DESCRIPTION
**Pull request by mbouron-gpsw**
The OpenGL documentation states:

> The pixel ownership test, the scissor test, dithering, and the buffer
writemasks affect the operation of glClear.

Without this patch, rendertargets can end up partially cleared if the
depth/stencil write masks are not reset to their default values.

Related to bdb5f389560d3e81a5a1019a6c0b74013f0fbc09.

### CODE REVIEW

File: tests/depth_stencil.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
```
Comment (ubitux) [inline tests/depth_stencil.py:1]: unneeded shebang

File: tests/depth_stencil.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+#
+# Copyright 2020 GoPro Inc.
```
Comment (ubitux) [inline tests/depth_stencil.py:3]: 2021 (or 2022)

File: tests/depth_stencil.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+from pynodegl_utils.toolbox.colors import COLORS
+
+
+def _render_quad(cfg, corner=(-1, -1, 0), width=(2, 0, 0), height=(0, 2, 0), color=(1, 1, 1, 1)):
+    quad = ngl.Quad(corner, width, height)
+    program = ngl.Program(vertex=cfg.get_vert('color'), fragment=cfg.get_frag('color'))
+    program.update_vert_out_vars(var_tex0_coord=ngl.IOVec2(), var_uvcoord=ngl.IOVec2())
+    render = ngl.Render(quad, program)
+    render.update_frag_resources(color=ngl.UniformVec4(value=color))
+    return render
```
Comment (ubitux) [inline tests/depth_stencil.py:35]: note to self: this needs to be adjusted in my branches if upstreamed before my other PRs

File: tests/meson.build [hunk 4]
```diff
@@ -174,6 +174,11 @@ foreach backend : backends
     endif
   endif
 
+  tests_depth_stencil = [
+    'depth',
+    'stencil',
+  ]
```
Comment (ubitux) [inline tests/meson.build:180]: Both tests are passing for me with your fix reverted. They're not supposed to be related?

**Review by ubitux — CHANGES_REQUESTED** [review]

File: tests/depth_stencil.py (hunk 1 above)
Comment (mbouron-gpsw) [reply tests/depth_stencil.py]: Removed.

File: tests/meson.build (hunk 4 above)
Comment (mbouron-gpsw) [reply tests/meson.build:180]: They are related, I updated them to use 2 keyframes. The tests are now failing if the first commit is reverted.
The issue can also be reproduced with `ngl-python -u tests/depth_stencil.py depth_stencil_depth` (`-u` is mandatory to reproduce the issue).

File: tests/depth_stencil.py (hunk 2 above)
Comment (mbouron-gpsw) [reply tests/depth_stencil.py]: Done.

File: tests/depth_stencil.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
+# Copyright 2022 GoPro Inc.
```
Comment (cboesch-gpsw) [inline tests/depth_stencil.py:1]: missing '#' at the beginning

**Review by cboesch-gpsw — CHANGES_REQUESTED** [review]

**Review by cboesch-gpsw — APPROVED** [review]

**Pull request merged**